nibabel/spatialimages.py

```diff
@@ -325,6 +326,23 @@ class SpatialImage(DataobjImage):
     ''' Template class for volumetric (3D/4D) images '''
     header_class = SpatialHeader
 
+    class Slicer(object):
+        def __init__(self, img):
+            self.img = img
+
+        def __getitem__(self, idx):
+            idx = self._validate(idx)
+            dataobj = self.img.dataobj[idx]
+            affine = self.img.slice_affine(idx)
+            return self.img.__class__(dataobj, affine, self.img.header)
```

Comment:
At present, this *always* returns a new image. That's consistent with the `array[:]` idiom, but in other cases we save memory by returning the same image if no change is made. So, if we want to check indices for the case where we're retrieving the whole dataobj, we can do that.

Reply:
I think returning a new image is fine.